<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>EditPianoSheetDialog</class>
 <widget class="QDialog" name="EditPianoSheetDialog">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>532</width>
    <height>513</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Piano data sheet</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <item>
    <widget class="QScrollArea" name="scrollArea">
     <property name="widgetResizable">
      <bool>true</bool>
     </property>
     <widget class="QWidget" name="scrollAreaWidgetContents">
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>0</y>
        <width>512</width>
        <height>462</height>
       </rect>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout_2">
       <item>
        <widget class="QGroupBox" name="pianoOperatingSiteInformation">
         <property name="title">
          <string>Piano operating site information</string>
         </property>
         <layout class="QFormLayout" name="pianoOperatingSiteInformationLayout">
          <property name="fieldGrowthPolicy">
           <enum>QFormLayout::AllNonFixedFieldsGrow</enum>
          </property>
          <item row="1" column="0">
           <widget class="QLabel" name="label_5">
            <property name="text">
             <string>Tuning location</string>
            </property>
           </widget>
          </item>
          <item row="1" column="1">
           <widget class="QLineEdit" name="tuningLocationLineEdit"/>
          </item>
          <item row="2" column="0">
           <widget class="QLabel" name="label_6">
            <property name="text">
             <string>Time of tuning</string>
            </property>
           </widget>
          </item>
          <item row="2" column="1">
           <layout class="QHBoxLayout" name="horizontalLayout">
            <item>
             <widget class="QDateTimeEdit" name="timeOfTuningDateTimeEdit">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="showGroupSeparator" stdset="0">
               <bool>false</bool>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QPushButton" name="timeOfTuningNowButton">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="text">
               <string>Now</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item row="0" column="0">
           <widget class="QLabel" name="label_7">
            <property name="text">
             <string>Concert pitch</string>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <layout class="QHBoxLayout" name="horizontalLayout_2">
            <item>
             <widget class="QDoubleSpinBox" name="concertPitchSpinBox">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="prefix">
               <string>A</string>
              </property>
              <property name="decimals">
               <number>1</number>
              </property>
              <property name="minimum">
               <double>400.000000000000000</double>
              </property>
              <property name="maximum">
               <double>452.000000000000000</double>
              </property>
              <property name="singleStep">
               <double>0.100000000000000</double>
              </property>
              <property name="value">
               <double>440.000000000000000</double>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QPushButton" name="defaultConcertPitchButton">
              <property name="text">
               <string>Default</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QGroupBox" name="pianoManufacturerInformation">
         <property name="title">
          <string>Piano manufacturer information</string>
         </property>
         <layout class="QFormLayout" name="pianoManufacturerInformationLayout">
          <property name="fieldGrowthPolicy">
           <enum>QFormLayout::AllNonFixedFieldsGrow</enum>
          </property>
          <item row="0" column="0">
           <widget class="QLabel" name="label">
            <property name="text">
             <string>Piano name</string>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="QLineEdit" name="nameLineEdit"/>
          </item>
          <item row="2" column="0">
           <widget class="QLabel" name="label_2">
            <property name="text">
             <string>Serial number</string>
            </property>
           </widget>
          </item>
          <item row="2" column="1">
           <widget class="QLineEdit" name="serialNumberLineEdit"/>
          </item>
          <item row="3" column="0">
           <widget class="QLabel" name="label_3">
            <property name="text">
             <string>Manufaction year</string>
            </property>
           </widget>
          </item>
          <item row="3" column="1">
           <widget class="QDateEdit" name="yearEdit">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="currentSection">
             <enum>QDateTimeEdit::YearSection</enum>
            </property>
            <property name="displayFormat">
             <string notr="true">yyyy</string>
            </property>
            <property name="calendarPopup">
             <bool>false</bool>
            </property>
           </widget>
          </item>
          <item row="4" column="0">
           <widget class="QLabel" name="label_4">
            <property name="text">
             <string>Production location</string>
            </property>
           </widget>
          </item>
          <item row="4" column="1">
           <widget class="QLineEdit" name="productionLocationLineEdit"/>
          </item>
          <item row="5" column="0">
           <widget class="QLabel" name="label_8">
            <property name="text">
             <string>Number of keys</string>
            </property>
           </widget>
          </item>
          <item row="5" column="1">
           <layout class="QHBoxLayout" name="horizontalLayout_3">
            <item>
             <widget class="QSpinBox" name="numberOfKeysSpinBox">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="minimum">
               <number>10</number>
              </property>
              <property name="value">
               <number>88</number>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QPushButton" name="defaultNumberOfKeysButton">
              <property name="text">
               <string>Default</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item row="6" column="0">
           <widget class="QLabel" name="label_9">
            <property name="text">
             <string>Key number of A</string>
            </property>
           </widget>
          </item>
          <item row="6" column="1">
           <layout class="QHBoxLayout" name="horizontalLayout_4">
            <item>
             <widget class="QSpinBox" name="keyNumberOfASpinBox">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="minimum">
               <number>10</number>
              </property>
              <property name="displayIntegerBase">
               <number>10</number>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QPushButton" name="defaultKeyNumberOfAButton">
              <property name="text">
               <string>Default</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item row="1" column="0">
           <widget class="QLabel" name="label_10">
            <property name="text">
             <string>Piano type</string>
            </property>
           </widget>
          </item>
          <item row="1" column="1">
           <widget class="QComboBox" name="pianoType">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <item>
             <property name="text">
              <string>Grand</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>Upright</string>
             </property>
            </item>
           </widget>
          </item>
          <item row="7" column="0">
           <widget class="QLabel" name="label_11">
            <property name="text">
             <string>Keys on bass bridge</string>
            </property>
           </widget>
          </item>
          <item row="7" column="1">
           <layout class="QHBoxLayout" name="horizontalLayout_5">
            <item>
             <widget class="QSpinBox" name="keysOnBassBridgeSpinBox">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="value">
               <number>28</number>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QPushButton" name="defaultKeysOnBassBridgeButton">
              <property name="text">
               <string>Default</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
         </layout>
        </widget>
       </item>
      </layout>
     </widget>
    </widget>
   </item>
   <item>
    <widget class="QDialogButtonBox" name="buttonBox">
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
     <property name="standardButtons">
      <set>QDialogButtonBox::Cancel|QDialogButtonBox::Ok</set>
     </property>
    </widget>
   </item>
  </layout>
 </widget>
 <tabstops>
  <tabstop>scrollArea</tabstop>
  <tabstop>concertPitchSpinBox</tabstop>
  <tabstop>defaultConcertPitchButton</tabstop>
  <tabstop>tuningLocationLineEdit</tabstop>
  <tabstop>timeOfTuningDateTimeEdit</tabstop>
  <tabstop>timeOfTuningNowButton</tabstop>
  <tabstop>nameLineEdit</tabstop>
  <tabstop>serialNumberLineEdit</tabstop>
  <tabstop>yearEdit</tabstop>
  <tabstop>productionLocationLineEdit</tabstop>
  <tabstop>numberOfKeysSpinBox</tabstop>
  <tabstop>defaultNumberOfKeysButton</tabstop>
  <tabstop>keyNumberOfASpinBox</tabstop>
  <tabstop>defaultKeyNumberOfAButton</tabstop>
 </tabstops>
 <resources/>
 <connections>
  <connection>
   <sender>buttonBox</sender>
   <signal>accepted()</signal>
   <receiver>EditPianoSheetDialog</receiver>
   <slot>accept()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>266</x>
     <y>503</y>
    </hint>
    <hint type="destinationlabel">
     <x>157</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonBox</sender>
   <signal>rejected()</signal>
   <receiver>EditPianoSheetDialog</receiver>
   <slot>reject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>334</x>
     <y>503</y>
    </hint>
    <hint type="destinationlabel">
     <x>286</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>timeOfTuningNowButton</sender>
   <signal>clicked()</signal>
   <receiver>EditPianoSheetDialog</receiver>
   <slot>onSetTuningTimeToNow()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>499</x>
     <y>157</y>
    </hint>
    <hint type="destinationlabel">
     <x>290</x>
     <y>203</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>numberOfKeysSpinBox</sender>
   <signal>valueChanged(int)</signal>
   <receiver>EditPianoSheetDialog</receiver>
   <slot>onNumberOfKeysChanged(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>393</x>
     <y>388</y>
    </hint>
    <hint type="destinationlabel">
     <x>296</x>
     <y>300</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>defaultConcertPitchButton</sender>
   <signal>clicked()</signal>
   <receiver>EditPianoSheetDialog</receiver>
   <slot>onDefaultConcertPitch()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>499</x>
     <y>84</y>
    </hint>
    <hint type="destinationlabel">
     <x>256</x>
     <y>270</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>defaultKeyNumberOfAButton</sender>
   <signal>clicked()</signal>
   <receiver>EditPianoSheetDialog</receiver>
   <slot>onDefaultKeyNumberOfA()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>499</x>
     <y>444</y>
    </hint>
    <hint type="destinationlabel">
     <x>256</x>
     <y>270</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>defaultNumberOfKeysButton</sender>
   <signal>clicked()</signal>
   <receiver>EditPianoSheetDialog</receiver>
   <slot>onDefaultNumberOfKeys()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>499</x>
     <y>406</y>
    </hint>
    <hint type="destinationlabel">
     <x>256</x>
     <y>270</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>defaultKeysOnBassBridgeButton</sender>
   <signal>clicked()</signal>
   <receiver>EditPianoSheetDialog</receiver>
   <slot>onDefaultKeysOnBassBridge()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>445</x>
     <y>441</y>
    </hint>
    <hint type="destinationlabel">
     <x>265</x>
     <y>256</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>onSetTuningTimeToNow()</slot>
  <slot>onNumberOfKeysChanged(int)</slot>
  <slot>onDefaultConcertPitch()</slot>
  <slot>onDefaultNumberOfKeys()</slot>
  <slot>onDefaultKeyNumberOfA()</slot>
  <slot>onDefaultKeysOnBassBridge()</slot>
 </slots>
</ui>
